### ISSUE TITLE
VHACD 4.0 not working through trimesh interface + dirty fix

### ISSUE THREAD

**Issue by alberthli** [mikedh/trimesh#1788]
I recently tried using [VHACD version 4.0](https://github.com/kmammou/v-hacd) and it doesn't work with the trimesh interface if I swap it out with the binary referenced in `vhacd.bash`. When I call `convex_decomposition(mesh)`, it simply returns an empty list. 

This is due to the revamped CLI. When calling `convex_decomposition(mesh, debug=True)`, we see the error
```
Failed to load valid mesh from wavefront OBJ file:--input
```
This is because in [this line](https://github.com/mikedh/trimesh/blob/b169bf6351cab36035ad26c085f9c994b271df10/trimesh/interfaces/vhacd.py#L43), the flags `--input`, `--output`, and `--log` are no longer valid and VHACD now always generates three files with the names `decomp.stl`, `decomp.mtl`, and `decomp.obj`.

A super quick and dirty fix to regain minimum functionality is as follows:

- In [this line](https://github.com/mikedh/trimesh/blob/b169bf6351cab36035ad26c085f9c994b271df10/trimesh/interfaces/vhacd.py#L43), replace `argstring = ' --input $MESH_0 --output $MESH_POST --log $SCRIPT'` with `argstring = ' $MESH_0'`.
- In [this line](https://github.com/mikedh/trimesh/blob/main/trimesh/interfaces/vhacd.py#L47) replace `argstring += ' --{} {}'.format(str(key),` with `argstring += ' -{} {}'.format(str(key),`
- In [this file](https://github.com/mikedh/trimesh/blob/main/trimesh/interfaces/generic.py), replace
```
mesh_results = exchange.load.load_mesh(self.mesh_post.name, **self.kwargs)
```
with
```
mesh_results = exchange.load.load_mesh("decomp.stl", **self.kwargs)
os.remove("decomp.stl")
os.remove("decomp.mtl")
os.remove("decomp.obj")
```
We can also comment out [this line](https://github.com/mikedh/trimesh/blob/main/trimesh/interfaces/generic.py#L111).

With these changes, VHACD 4.0 works and we can pass in the new flags. I didn't open a PR because I'm sure there are much cleaner ways to implement the fix, but let me know if you would like me to.

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

For reference, here's the usage output for version 4.0:
```
Usage: TestVHACD <wavefront.obj> (options)

-h <n>                  : Maximum number of output convex hulls. Default is 32
-r <voxelresolution>    : Total number of voxels to use. Default is 100,000
-e <volumeErrorPercent> : Volume error allowed as a percentage. Default is 1%. Valid range is 0.001 to 10
-d <maxRecursionDepth>  : Maximum recursion depth. Default value is 10.
-s <true/false>         : Whether or not to shrinkwrap output to source mesh. Default is true.
-f <fillMode>           : Fill mode. Default is 'flood', also 'surface' and 'raycast' are valid.
-v <maxHullVertCount>   : Maximum number of vertices in the output convex hull. Default value is 64
-a <true/false>         : Whether or not to run asynchronously. Default is 'true'
-l <minEdgeLength>      : Minimum size of a voxel edge. Default value is 2 voxels.
-p <true/false>         : If false, splits hulls in the middle. If true, tries to find optimal split plane location. False by default.
-o <obj/stl/usda>       : Export the convex hulls as a series of wavefront OBJ files, STL files, or a single USDA.
-g <true/false>         : If set to false, no logging will be displayed.
```
Here's the usage output for version 2.3:
```
V-HACD V2.3
Syntax: testVHACD [options] --input infile.obj --output outfile.wrl --log logfile.txt

Options:
       --input                     Wavefront .obj input file name
       --output                    VRML 2.0 output file name
       --log                       Log file name
       --resolution                Maximum number of voxels generated during the voxelization stage (default=100,000, range=10,000-16,000,000)
       --maxhulls                  Maximum number of convex hulls to produce.
       --concavity                 Maximum allowed concavity (default=0.0025, range=0.0-1.0)
       --planeDownsampling         Controls the granularity of the search for the "best" clipping plane (default=4, range=1-16)
       --convexhullDownsampling    Controls the precision of the convex-hull generation process during the clipping plane selection stage (default=4, range=1-16)
       --alpha                     Controls the bias toward clipping along symmetry planes (default=0.05, range=0.0-1.0)
       --beta                      Controls the bias toward clipping along revolution axes (default=0.05, range=0.0-1.0)
       --gamma                     Controls the maximum allowed concavity during the merge stage (default=0.00125, range=0.0-1.0)
       --delta                     Controls the bias toward maximaxing local concavity (default=0.05, range=0.0-1.0)
       --pca                       Enable/disable normalizing the mesh before applying the convex decomposition (default=0, range={0,1})
       --mode                      0: voxel-based approximate convex decomposition, 1: tetrahedron-based approximate convex decomposition (default=0, range={0,1})
       --maxNumVerticesPerCH       Controls the maximum number of triangles per convex-hull (default=64, range=4-1024)
       --minVolumePerCH            Controls the adaptive sampling of the generated convex-hulls (default=0.0001, range=0.0-0.01)
       --convexhullApproximation   Enable/disable approximation when computing convex-hulls (default=1, range={0,1})
       --oclAcceleration           Enable/disable OpenCL acceleration (default=0, range={0,1})
       --oclPlatformID             OpenCL platform id (default=0, range=0-# OCL platforms)
       --oclDeviceID               OpenCL device id (default=0, range=0-# OCL devices)
       --help                      Print usage

Examples:
       testVHACD.exe --input bunny.obj --output bunny_acd.wrl --log log.txt

+ OpenCL (OFF)
+ OpenMP (ON)
+ Parameters
	 input                                       
	 resolution                                  100000
	 max. concavity                              0.001
	 plane down-sampling                         4
	 convex-hull down-sampling                   4
	 alpha                                       0.05
	 beta                                        0.05
	 maxhulls                                    1024
	 pca                                         0
	 mode                                        0
	 max. vertices per convex-hull               64
	 min. volume to add vertices to convex-hulls 0.0001
	 convex-hull approximation                   1
	 OpenCL acceleration                         1
	 OpenCL platform ID                          0
	 OpenCL device ID                            0
	 output                                      output.wrl
	 log                                         log.txt
+ Load mesh
```

**Comment by musicinmybrain** [comment]
I just packaged V-HACD 4.1 for Fedora Linux with the intent of enabling this optional functionality, then discovered this issue. If you have any thoughts on how you would like to deal with the change in the command-line interface, I’m happy to help craft a PR.

**Comment by chlongv** [comment]
I stumbled upon the same issue: I had packaged `TestVHACD` built from the current git master in my Docker image and noticed that the command line arguments had changed.

It would be nice to update this as VHACD version 4 has been around for a few months and is now the officially supported version. The only problem is that `TestVHACD` generates 3 files (`decomp.mtl`, `decomp.obj`, and `decomp.stl`), and one of these 3 files has  to be used to load the mesh from (and the files must be cleaned) ... this does not integrate well with `MeshScript`.

**Comment by musicinmybrain** [comment]
It looks like [4.0.0rc1](https://github.com/mikedh/trimesh/pull/2039) drops the dependency on `TestVHACD` in favor of https://pypi.org/project/pyVHACD/, which is [currently built with](https://github.com/thomwolf/pyVHACD/blob/main/src/pyVHACD/VHACD.h) a bundled copy of V-HACD 4.0, but which builds just fine with an external copy of V-HACD 4.1 (which is how I will package it for Fedora).

**Comment by alberthli** [comment]
This issue was resolved in the recent `trimesh 4.0.0` release.
